src/blk/kernel/KernelDevice.cc

```diff
@@ -88,11 +88,26 @@ KernelDevice::KernelDevice(CephContext* cct, aio_callback_t cb, void *cbpriv, ai
     }
     io_queue = std::make_unique<aio_queue_t>(iodepth);
   }
+
+  char name[128];
+  sprintf(name, "blk-kernel-device_%s", type);
+  PerfCountersBuilder b(cct, name,
+                       l_blk_kernel_device_first, l_blk_kernel_device_last);
+  b.set_prio_default(PerfCountersBuilder::PRIO_USEFUL);
+  b.add_u64_counter(l_blk_kernel_device_discard_op, "discard_op",
```

Comment:
Can we also track amount of ops ever queued to the device? Hence the delta between it and this parameter will show the queue length?

Reply:
We just need to configure this amount as rate in grafana, and the result we see is the current queue length.